packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -0,0 +1,167 @@
+import { makeStyles, flexRowCentered } from 'app/styles'
+import { useThemeColors } from 'app/utils/theme'
+
+import { TrackDetailsTile } from '../track-details-tile'
+
+const { getTrack } = cacheTracksSelectors
+
+const PREMIUM_TRACK_PURCHASE_MODAL_NAME = 'PremiumTrackPurchase'
+
+const messages = {
+  title: 'Complete Purchase',
+  summary: 'Summary',
+  artistCut: 'Artist Cut',
+  audiusCut: 'Audius Cut',
+  alwaysZero: 'Always $0',
+  youPay: 'You Pay',
+  price: (price: string) => `$${price}`,
+  payToUnlock: 'Pay-To-Unlock',
+  disclaimer:
+    'By clicking on "Buy", you agree to our Terms of Use. Your purchase will be made in USDC via 3rd party payment provider. Additional payment provider fees may apply. Any remaining USDC balance in your Audius wallet will be applied to this transaction. Once your payment is confirmed, your premium content will be unlocked and available to stream.',
+  buy: (price: string) => `Buy $${price}`
+}
+
+const useStyles = makeStyles(({ spacing, typography, palette }) => ({
+  drawer: {
+    paddingVertical: spacing(6),
+    paddingHorizontal: spacing(4),
+    gap: spacing(6),
+    backgroundColor: palette.white
+  },
+  titleContainer: {
+    ...flexRowCentered(),
+    gap: spacing(2),
+    marginBottom: spacing(2),
+    alignSelf: 'center'
+  },
+  title: {
+    fontSize: typography.fontSize.xl,
+    fontFamily: typography.fontByWeight.heavy,
+    color: palette.neutralLight2,
+    textTransform: 'uppercase',
+    lineHeight: typography.fontSize.xl * 1.25
+  },
+  trackTileContainer: {
+    ...flexRowCentered(),
+    borderColor: palette.neutralLight8,
+    borderWidth: 1,
+    borderRadius: spacing(2),
+    backgroundColor: palette.neutralLight10
+  },
+  summaryContainer: {
+    borderColor: palette.neutralLight8,
+    borderWidth: 1,
+    borderRadius: spacing(1)
+  },
+  summaryRow: {
+    ...flexRowCentered(),
+    justifyContent: 'space-between',
+    paddingVertical: spacing(3),
+    paddingHorizontal: spacing(6),
+    borderBottomColor: palette.neutralLight8,
+    borderBottomWidth: 1
+  },
+  lastRow: {
+    borderBottomWidth: 0
+  },
+  greyRow: {
+    backgroundColor: palette.neutralLight10
+  },
+  summaryTitle: {
+    letterSpacing: 1
+  },
+  payToUnlockTitleContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    gap: spacing(2),
+    marginBottom: spacing(2)
+  }
+}))
+
+export const PremiumTrackPurchaseDrawer = () => {
+  const styles = useStyles()
+  const { neutralLight2, specialLightGreen1 } = useThemeColors()
+  const { data } = useDrawer('PremiumTrackPurchase')
+  const { trackId } = data
+  const track = useSelector((state) => getTrack(state, { id: trackId }))
+
+  const handleConfirmPress = useCallback(() => {
+    console.log('buy button pressed')
+  }, [])
+
+  const { premium_conditions: premiumConditions } = track ?? {}
+
+  if (!track || !isPremiumContentUSDCPurchaseGated(premiumConditions))
```

Comment:
Copying over convo from previous PR:
@schottra :
I would normally argue that this component should not even be rendered if the conditions here aren't true (prefer that components receive only valid data). But I would understand if that's a difficult thing to check at the layer where we render this component.
@dharit-tan :
yeah agree. i think the difficulty is since this is a drawer, all we pass to the drawer is a trackId and so in this component we don't know whether it's a USDC-gated track or if the track even exists in the cache. I suppose we could pass in the premiumConditions object to the drawer, but that seems to go against the "select everything at the leaf" mental model. Open to ideas here.

Reply:
Ah interesting. I guess I'm hoping we don't launch the drawer at all if we don't have the right data available, even if we eventually only pass the trackId to the drawer.

So, in the case where some of these things conditions are for some reason false, we're going to see an empty drawer? Wonder if we should render some kind of error in this case instead of null...

Either way, probably okay to leave for now.

Reply:
i hear you. the 1 place we open the drawer, we do guarantee that we have the correct data, so for now we're good.
packages/mobile/src/components/lineup-tile/LineupTileAccessStatus.tsx line 80.